docs/modules/_fps_config_.md
[hkqr-fps](../README.md) › ["FPS/config"](_fps_config_.md)

# Module: "FPS/config"

## Index

### Interfaces

* [ICODE](../interfaces/_fps_config_.icode.md)

### Type aliases

* [COUNTRY](_fps_config_.md#country)
* [CURRENCY](_fps_config_.md#currency)
* [LANGUAGE](_fps_config_.md#language)
* [MERCHANT](_fps_config_.md#merchant)
* [MERCHANT_INFO](_fps_config_.md#merchant_info)
* [PARTICIPANTS](_fps_config_.md#participants)

### Variables

* [COUNTRY](_fps_config_.md#const-country)
* [CURRENCY](_fps_config_.md#const-currency)
* [LANGUAGE](_fps_config_.md#const-language)
* [MERCHANT](_fps_config_.md#const-merchant)
* [PARTICIPANT](_fps_config_.md#const-participant)

## Type aliases

###  COUNTRY

Ƭ **COUNTRY**: *keyof typeof COUNTRY*

Defined in FPS/config.ts:68

___

###  CURRENCY

Ƭ **CURRENCY**: *keyof typeof CURRENCY*

Defined in FPS/config.ts:69

___

###  LANGUAGE

Ƭ **LANGUAGE**: *keyof typeof LANGUAGE*

Defined in FPS/config.ts:70

___

###  MERCHANT

Ƭ **MERCHANT**: *typeof MERCHANT*

Defined in FPS/config.ts:71

___

###  MERCHANT_INFO

Ƭ **MERCHANT_INFO**: *object*

Defined in FPS/config.ts:74

#### Type declaration:

* **extra**? : *[VALID_OBJECT](_lib_constant_.md#valid_object)*

* **language**: *[LANGUAGE](_fps_config_.md#language)*

* **merchantCity**? : *string*

* **merchantName**: *string*

___

###  PARTICIPANTS

Ƭ **PARTICIPANTS**: *keyof typeof PARTICIPANT*

Defined in FPS/config.ts:72

## Variables

### `Const` COUNTRY

• **COUNTRY**: *object* = {
    CN: "cn",
    GB: "gb",
    HK: "hk",
    MO: "mo",
    TW: "tw",
    US: "us",
} as const

Defined in FPS/config.ts:4

#### Type declaration:

* **CN**: *"cn"* = "cn"

* **GB**: *"gb"* = "gb"

* **HK**: *"hk"* = "hk"

* **MO**: *"mo"* = "mo"

* **TW**: *"tw"* = "tw"

* **US**: *"us"* = "us"

___

### `Const` CURRENCY

• **CURRENCY**: *object* = {
    CNY: "156",
    HKD: "344",
    TWD: "901",
    USD: "840",
} as const

Defined in FPS/config.ts:12

#### Type declaration:

* **CNY**: *"156"* = "156"

* **HKD**: *"344"* = "344"

* **TWD**: *"901"* = "901"

* **USD**: *"840"* = "840"

___

### `Const` LANGUAGE

• **LANGUAGE**: *object* = {
    EN: "en",
    ZH: "zh",
} as const

Defined in FPS/config.ts:18

#### Type declaration:

* **EN**: *"en"* = "en"

* **ZH**: *"zh"* = "zh"

___

### `Const` MERCHANT

• **MERCHANT**: *"26"* = "26"

Defined in FPS/config.ts:22

___

### `Const` PARTICIPANT

• **PARTICIPANT**: *object* = {
    "003": "STANDARD CHARTERED BANK (HONG KONG) LIMITED",
    "004": "The Hongkong and Shanghai Banking Corporation Limited",
    "009": "China Construction Bank (Asia) Corporation Limited",
    "012": "Bank of China (Hong Kong) Limited",
    "014": "Bank of China (Hong Kong) Limited",
    "015": "The Bank of East Asia, Limited",
    "016": "DBS Bank (Hong Kong) Ltd.",
    "019": "Bank of China (Hong Kong) Limited",
    "020": "CMB Wing Lung Bank Limited",
    "024": "Hang Seng Bank Ltd.",
    "025": "Shanghai Commercial Bank Limited",
    "026": "Bank of China (Hong Kong) Limited",
    "027": "Bank of Communications Co., Ltd. Hong Kong Branch",
    "028": "Public Bank (Hong Kong) Limited",
    "029": "INDUSTRIAL AND COMMERCIAL BANK OF CHINA (ASIA) LIMITED",
    "030": "Bank of China (Hong Kong) Limited",
    "031": "Bank of China (Hong Kong) Limited",
    "033": "Bank of China (Hong Kong) Limited",
    "035": "OCBC Wing Hang Bank Limited",
    "036": "Bank of China (Hong Kong) Limited",
    "038": "Tai Yau Bank Limited",
    "039": "Chiyu Banking Corporation Limited",
    "040": "Dah Sing Bank, Limited",
    "041": "Chong Hing Bank Limited",
    "043": "Nanyang Commercial Bank, Limited",
    "044": "OCBC Wing Hang Bank Limited",
    "061": "TAI SANG BANK LTD.",
    "064": "Bank of China (Hong Kong) Limited",
    "070": "Bank of China (Hong Kong) Limited",
    "128": "Fubon Bank (Hong Kong) Limited",
    "250": "Citibank (Hong Kong) Limited",
    "929": "K & R INTERNATIONAL LIMITED",
    "930": "EPAYLINKS TECHNOLOGY CO., LIMITED",
    "931": "WeChat Pay Hong Kong Limited",
    "933": "33 Financial Services Limited",
    "934": "UniCard Solution Limited",
    "935": "HKT Payment Limited",
    "947": "TNG (Asia) Limited",
    "948": "Alipay Financial Services (HK) Limited",
    "949": "Octopus Cards Limited",
    "952": "Autotoll Limited",
    "954": "PayMe",
} as const

Defined in FPS/config.ts:23

#### Type declaration:

* **003**: *"STANDARD CHARTERED BANK (HONG KONG) LIMITED"* = "STANDARD CHARTERED BANK (HONG KONG) LIMITED"

* **004**: *"The Hongkong and Shanghai Banking Corporation Limited"* = "The Hongkong and Shanghai Banking Corporation Limited"

* **009**: *"China Construction Bank (Asia) Corporation Limited"* = "China Construction Bank (Asia) Corporation Limited"

* **012**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **014**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **015**: *"The Bank of East Asia, Limited"* = "The Bank of East Asia, Limited"

* **016**: *"DBS Bank (Hong Kong) Ltd."* = "DBS Bank (Hong Kong) Ltd."

* **019**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **020**: *"CMB Wing Lung Bank Limited"* = "CMB Wing Lung Bank Limited"

* **024**: *"Hang Seng Bank Ltd."* = "Hang Seng Bank Ltd."

* **025**: *"Shanghai Commercial Bank Limited"* = "Shanghai Commercial Bank Limited"

* **026**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **027**: *"Bank of Communications Co., Ltd. Hong Kong Branch"* = "Bank of Communications Co., Ltd. Hong Kong Branch"

* **028**: *"Public Bank (Hong Kong) Limited"* = "Public Bank (Hong Kong) Limited"

* **029**: *"INDUSTRIAL AND COMMERCIAL BANK OF CHINA (ASIA) LIMITED"* = "INDUSTRIAL AND COMMERCIAL BANK OF CHINA (ASIA) LIMITED"

* **030**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **031**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **033**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **035**: *"OCBC Wing Hang Bank Limited"* = "OCBC Wing Hang Bank Limited"

* **036**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **038**: *"Tai Yau Bank Limited"* = "Tai Yau Bank Limited"

* **039**: *"Chiyu Banking Corporation Limited"* = "Chiyu Banking Corporation Limited"

* **040**: *"Dah Sing Bank, Limited"* = "Dah Sing Bank, Limited"

* **041**: *"Chong Hing Bank Limited"* = "Chong Hing Bank Limited"

* **043**: *"Nanyang Commercial Bank, Limited"* = "Nanyang Commercial Bank, Limited"

* **044**: *"OCBC Wing Hang Bank Limited"* = "OCBC Wing Hang Bank Limited"

* **061**: *"TAI SANG BANK LTD."* = "TAI SANG BANK LTD."

* **064**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **070**: *"Bank of China (Hong Kong) Limited"* = "Bank of China (Hong Kong) Limited"

* **128**: *"Fubon Bank (Hong Kong) Limited"* = "Fubon Bank (Hong Kong) Limited"

* **250**: *"Citibank (Hong Kong) Limited"* = "Citibank (Hong Kong) Limited"

* **929**: *"K & R INTERNATIONAL LIMITED"* = "K & R INTERNATIONAL LIMITED"

* **930**: *"EPAYLINKS TECHNOLOGY CO., LIMITED"* = "EPAYLINKS TECHNOLOGY CO., LIMITED"

* **931**: *"WeChat Pay Hong Kong Limited"* = "WeChat Pay Hong Kong Limited"

* **933**: *"33 Financial Services Limited"* = "33 Financial Services Limited"

* **934**: *"UniCard Solution Limited"* = "UniCard Solution Limited"

* **935**: *"HKT Payment Limited"* = "HKT Payment Limited"

* **947**: *"TNG (Asia) Limited"* = "TNG (Asia) Limited"

* **948**: *"Alipay Financial Services (HK) Limited"* = "Alipay Financial Services (HK) Limited"

* **949**: *"Octopus Cards Limited"* = "Octopus Cards Limited"

* **952**: *"Autotoll Limited"* = "Autotoll Limited"

* **954**: *"PayMe"* = "PayMe"


src/HKQR/config.ts
import Event from "../lib/event";

import ISO_COUNTRIES from "../lib/iso/country";
import ISO_CURRENCIES from "../lib/iso/currency";
import ISO_LANGUAGES from "../lib/iso/language";
import ISO_MERCHANT_CATEGORIES from "../lib/iso/merchant_category";
import * as LIST_MERCHANTS from "../lib/merchantID";
import LIST_PARTICIPANTS from "../lib/participant";
import { POINT_OF_INITIATION } from "../lib/constant";

export const ISO_COUNTRY = ISO_COUNTRIES;
export const ISO_CURRENCY = ISO_CURRENCIES;
export const ISO_LANGUAGE = ISO_LANGUAGES;
export const ISO_MERCHANT_CATEGORY = ISO_MERCHANT_CATEGORIES;
export const MERCHANT = LIST_MERCHANTS;
export const PARTICIPANT = LIST_PARTICIPANTS;

export type VALID_ID = "00" | "01" | "02" | "03" | "04" | "05" | "06" | "07" | "08" | "09" | "10" | "11" | "12" | "13" | "14" | "15" | "16" | "17" | "18" | "19" | "20" | "21" | "22" | "23" | "24" | "25" | "26" | "27" | "28" | "29" | "30" | "31" | "32" | "33" | "34" | "35" | "36" | "37" | "38" | "39" | "40" | "41" | "42" | "43" | "44" | "45" | "46" | "47" | "48" | "49" | "50" | "51" | "52" | "53" | "54" | "55" | "56" | "57" | "58" | "59" | "60" | "61" | "62" | "63" | "64" | "65" | "66" | "67" | "68" | "69" | "70" | "71" | "72" | "73" | "74" | "75" | "76" | "77" | "78" | "79" | "80" | "81" | "82" | "83" | "84" | "85" | "86" | "87" | "88" | "89" | "90" | "91" | "92" | "93" | "94" | "95" | "96" | "97" | "98" | "99";
export type VALID_OBJECT = {
    [key in VALID_ID]?: string;
}

// Merchant Accounts | ID: [02..51]
export type MERCHANT_ACCOUNT_VISA = typeof MERCHANT.ACCOUNT_VISA[number];
export type MERCHANT_ACCOUNT_MASTER = typeof MERCHANT.ACCOUNT_MASTER[number];
export type MERCHANT_ACCOUNT_EMVCO = typeof MERCHANT.ACCOUNT_EMVCO[number];
export type MERCHANT_ACCOUNT_DISCOVER = typeof MERCHANT.ACCOUNT_DISCOVER[number];
export type MERCHANT_ACCOUNT_AMEX = typeof MERCHANT.ACCOUNT_AMEX[number];
export type MERCHANT_ACCOUNT_JCB = typeof MERCHANT.ACCOUNT_JCB[number];
export type MERCHANT_ACCOUNT_UNION = typeof MERCHANT.ACCOUNT_UNION[number];
export type MERCHANT_ACCOUNT_FPS = typeof MERCHANT.ACCOUNT_FPS[number];
export type MERCHANT_ACCOUNT_WG = typeof MERCHANT.ACCOUNT_WG[number];
export type MERCHANT_ACCOUNT_PO = typeof MERCHANT.ACCOUNT_PO[number];
export type MERCHANT_ACCOUNTS = MERCHANT_ACCOUNT_VISA | MERCHANT_ACCOUNT_MASTER | MERCHANT_ACCOUNT_EMVCO | MERCHANT_ACCOUNT_DISCOVER | MERCHANT_ACCOUNT_AMEX | MERCHANT_ACCOUNT_JCB | MERCHANT_ACCOUNT_UNION | MERCHANT_ACCOUNT_FPS | MERCHANT_ACCOUNT_WG | MERCHANT_ACCOUNT_PO;

// ISO Data
export type COUNTRY = keyof typeof ISO_COUNTRY;
export type CURRENCY = keyof typeof ISO_CURRENCY;
export type LANGUAGE = keyof typeof ISO_LANGUAGE;
export type MERCHANT_CATEGORY = keyof typeof ISO_MERCHANT_CATEGORIES;
export type PARTICIPANTS = keyof typeof PARTICIPANT;

export type MERCHANT_INFO = {
    language: LANGUAGE; // ID: 00
    merchantName: string; // ID: 01
    merchantCity?: string; // ID: 02
    extra?: VALID_OBJECT; // ID: [03..99]
};

export interface ICODE {
    parse(x: string): Event;
    extract(x: string): Event;
    generate(): Event;
    isStaticQRCode(): boolean;
    getInitiationPoint(): Event;
    setInitiationPoint(x: POINT_OF_INITIATION): Event;
    getMerchantAccountId(): Event;
    setMerchantAccountId(x: MERCHANT_ACCOUNTS): Event;
    getCountryCode(): Event;
    setCountryCode(x: COUNTRY): Event;
    getMerchantAccount(): Event;
    setMerchantAccount(x: VALID_OBJECT): Event;
    getUniqueIdentifier(): Event;
    setUniqueIdentifier(x: string): Event;
    getMerchantAccountParticipantCode(toName: boolean): Event;
    setMerchantAccountParticipantCode(x: PARTICIPANTS): Event;
    getMerchantAccountFPSId(): Event;
    setMerchantAccountFPSId(x: number | string): Event;
    getMerchantAccountMobile(): Event;
    setMerchantAccountMobile(x: number | string): Event;
    getMerchantAccountEmail(): Event;
    setMerchantAccountEmail(x: string): Event;
    getPaymentNetwork(): Event;
    setPaymentNetwork(id: VALID_ID, x: string): Event;
    getMerchantName(): Event;
    setMerchantName(x: string): Event;
    getMerchantCity(): Event;
    setMerchantCity(x: string): Event;
    getPostalCode(): Event;
    setPostalCode(x: string): Event;
    getMerchantCategory(toName: boolean): Event;
    setMerchantCategory(x: MERCHANT_CATEGORY): Event;
    getTransactionCurrency(toCode: boolean): Event;
    setTransactionCurrency(x: CURRENCY): Event;
    getTransactionAmount(toNumber: boolean): Event;
    setTransactionAmount(x: number | string, fraction: number | boolean): Event;
    getConvenienceFeeAmount(toNumber: boolean): Event;
    setConvenienceFeeAmount(x: number | string, fraction: number | boolean): Event;
    getConvenienceFeePercent(toNumber: boolean): Event;
    setConvenienceFeePercent(x: number | string, fraction: number | boolean): Event;
    getAdditionalInfo(): Event;
    setAdditionalInfo(x: VALID_OBJECT): Event;
    getBillNumber(): Event;
    setBillNumber(x: string): Event;
    getCustomerMobileNumber(): Event;
    setCustomerMobileNumber(x: string): Event;
    getStoreLabel(): Event;
    setStoreLabel(x: string): Event;
    getLoyaltyNumber(): Event;
    setLoyaltyNumber(x: string): Event;
    getReferenceLabel(): Event;
    setReferenceLabel(x: string): Event;
    getCustomerLabel(): Event;
    setCustomerLabel(x: string): Event;
    getTerminalLabel(): Event;
    setTerminalLabel(x: string): Event;
    getTransactionPurpose(): Event;
    setTransactionPurpose(x: string): Event;
    getCustomerDataRequest(): Event;
    setCustomerDataRequest(x: string): Event;
    getExtraAdditionalData(): Event;
    setExtraAdditionalData(id: VALID_ID, x: string): Event;
    getMerchantInfo(): Event;
    setMerchantInfo(x: VALID_OBJECT): Event;
    getLanguagePreference(): Event;
    setLanguagePreference(x: LANGUAGE): Event;
    getLocalizedMerchantName(): Event;
    setLocalizedMerchantName(x: string): Event;
    getLocalizedMerchantCity(): Event;
    setLocalizedMerchantCity(x: string): Event;
    getExtraLocalizedData(): Event;
    setExtraLocalizedData(id: VALID_ID, x: string): Event;
}